elastic4s-client-http4s/src/test/scala/com/sksamuel/elastic4s/http4s/Http4sRequestHttpClientTest.scala

```diff
@@ -0,0 +1,35 @@
+package com.sksamuel.elastic4s.http4s
+
+import cats.effect.IO
+import cats.effect.unsafe.implicits.global
+import com.sksamuel.elastic4s.testkit.DockerTests
+import com.sksamuel.elastic4s.{Authentication, CommonRequestOptions, ElasticClient, ElasticNodeEndpoint}
+import org.http4s.ember.client.EmberClientBuilder
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+class Http4sRequestHttpClientTest extends AnyFlatSpec with Matchers with DockerTests {
+  private val http4s = EmberClientBuilder.default[IO].build.allocated.unsafeRunSync()._1
+  private val http4sClient = Http4sClient.usingIO(
+    http4s,
+    ElasticNodeEndpoint("http", elasticHost, elasticPort.toInt, None),
+  )
+  override val client: ElasticClient = ElasticClient(http4sClient)
+
+  "Http4sRequestHttpClient" should "be able to call elasticsearch" in {
+    client.execute {
+      catHealth()
+    }.await.result.status shouldBe "green"
```

Comment:
Looks like some times the test gives back `yellow`, which I think for this test is ok.

Reply:
@Philippus fixed. I think it would be a good idea to extract a common test suite for all http clients (since they all can be tested in a single way), will try to do it in a separate PR, and then more tests can be added